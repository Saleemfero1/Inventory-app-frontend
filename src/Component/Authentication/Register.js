import * as React from "react";
import Avatar from "@mui/material/Avatar";
import Button from "@mui/material/Button";
import CssBaseline from "@mui/material/CssBaseline";
import TextField from "@mui/material/TextField";
import FormControlLabel from "@mui/material/FormControlLabel";
import Checkbox from "@mui/material/Checkbox";
import Link from "@mui/material/Link";
import Grid from "@mui/material/Grid";
import Box from "@mui/material/Box";
import LockOutlinedIcon from "@mui/icons-material/LockOutlined";
import Typography from "@mui/material/Typography";
import Container from "@mui/material/Container";
import { createTheme, ThemeProvider } from "@mui/material/styles";
import { useState } from "react";
import AuthServices from "../../Service/AuthServices";
import { toast, ToastContainer } from "react-toastify";
import { useNavigate } from "react-router-dom";
import { blue } from "@mui/material/colors";

function Copyright(props) {
  return (
    <Typography variant="body2" color=" #6a7ae4" align="center" {...props}>
      {"Copyright Â© "}
      <Link color="inherit" href="https://mui.com/">
        TupleInventory Management
      </Link>{" "}
      {new Date().getFullYear()}
      {"."}
    </Typography>
  );
}

const defaultTheme = createTheme();

export default function Register() {
  const navigate = useNavigate();
  const [user, setUser] = useState({
    username: "",
    userEmail: "",
    organizationId: "",
    password: "",
  });
  const [EmailError, setEmailError] = useState("Email Adress");

  const handleSubmit = (event) => {
    event.preventDefault();
  };

  const onChangeUserName = (event) => {
    setUser({ ...user, username: event.target.value });
  };

  const onChangeUserEmail = (event) => {
    setUser({ ...user, userEmail: event.target.value });
  };

  const onChangeOrganizationId = (event) => {
    setUser({ ...user, organizationId: event.target.value });
  };

  const onChangePassword = (event) => {
    setUser({ ...user, password: event.target.value });
  };

  const registerUser = (event) => {
    event.preventDefault();
    AuthServices.registerUser(user)
      .then((response) => {
        toast.success(response.data, {
          onClose: () => navigate("/signin"),
        });
      })
      .catch((err) => {
        toast.error(err.response.data.message);
      });
  };

  return (
    <ThemeProvider theme={defaultTheme}>
      <ToastContainer position="bottom-center" />
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Box
          sx={{
            marginTop: 8,
            display: "flex",
            backgroundColor: blue,
            flexDirection: "column",
            alignItems: "center",
          }}
        >
          <Avatar sx={{ m: 1, bgcolor: "#6a7ae4" }}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Sign up
          </Typography>
          <Box
            component="form"
            noValidate
            onSubmit={registerUser}
            sx={{ mt: 3 }}
          >
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <TextField
                  autoComplete="given-name"
                  name="username"
                  required
                  fullWidth
                  id="username"
                  label="User Name"
                  value={user.username}
                  onChange={onChangeUserName}
                  autoFocus
                />
              </Grid>

              <Grid item xs={12}>
                <TextField
                  required
                  fullWidth
                  id="email"
                  label="Email Address"
                  name="email"
                  autoComplete="email"
                  value={user.userEmail}
                  onChange={onChangeUserEmail}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  required
                  fullWidth
                  id="organizationId"
                  label="Organization Id"
                  name="OrganizationId"
                  autoComplete="Organization"
                  value={user.organizationId}
                  onChange={onChangeOrganizationId}
                />
              </Grid>
              <Grid item xs={12}>
                <TextField
                  required
                  fullWidth
                  name="password"
                  label="Password"
                  type="password"
                  id="password"
                  autoComplete="new-password"
                  value={user.password}
                  onChange={onChangePassword}
                />
              </Grid>
              <Grid item xs={12}>
                <FormControlLabel
                  control={
                    <Checkbox value="allowExtraEmails" color="primary" />
                  }
                  label="I want to receive inspiration, marketing promotions and updates via email."
                />
              </Grid>
            </Grid>
            <Button
              type="submit"
              fullWidth
              variant="contained"
              sx={{ mt: 3, mb: 2, bgcolor: "#6a7ae4" }}
            >
              Sign Up
            </Button>
            <Grid container justifyContent="flex-end">
              <Grid item>
                <Link variant="body2" to="/signin" color={"#6a7ae4"}>
                  Already have an account? Sign in
                </Link>
              </Grid>
            </Grid>
          </Box>
        </Box>
        <Copyright sx={{ mt: 5 }} />
      </Container>
    </ThemeProvider>
  );
}
