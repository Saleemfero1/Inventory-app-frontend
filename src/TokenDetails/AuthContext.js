import React, { createContext, useState } from "react";

export const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  const [isAuthenticated, setIsAuthenticated] = useState(true);
  const [token, setToken] = useState(sessionStorage.getItem("token"));
  const [userName, setUser] = useState(sessionStorage.getItem("username"));
  const [dashData, setDashData] = useState([]);
  const [itemData, setItemData] = useState([]);
  const [locationData, setLocationData] = useState([]);
  const [supplyData, setSupplyData] = useState([]);
  const [demandData, setDemandData] = useState([]);
  const [thresholdData, setThresholdData] = useState([]);
  const [totalItem, setTotalItem] = useState(0);
  const [totalActiveItem, setTotalActiveItem] = useState(0);
  const [totalSupply, setTotalSupply] = useState(0);
  const [totalDemand, setTotalDemand] = useState(0);
  const [digg, setDigg] = useState({});
  const [totalLocation, setTotalLocation] = useState(0);
  const [TransactionData, setTransactionData] = useState([]);
  const [lowAndHighStock, setlowAndHighStock] = useState([]);
  const [TrendingItems, SetTrendingItems] = useState([]);
  const updateToken = (newToken, newUser) => {
    if (newToken) {
      setToken(newToken);
    }
    setUser(newUser);
  };

  return (
    <AuthContext.Provider
      value={{
        token,
        updateToken,
        userName,
        dashData,
        setDashData,
        setItemData,
        itemData,
        locationData,
        setLocationData,
        supplyData,
        setSupplyData,
        demandData,
        setDemandData,
        thresholdData,
        setThresholdData,
        isAuthenticated,
        setIsAuthenticated,
        totalItem,
        setTotalItem,
        totalActiveItem,
        setTotalActiveItem,
        setTotalSupply,
        totalSupply,
        setTotalDemand,
        totalDemand,
        totalLocation,
        setTotalLocation,
        digg,
        setDigg,
        TransactionData,
        setTransactionData,
        lowAndHighStock,
        setlowAndHighStock,
        TrendingItems,
        SetTrendingItems,
      }}
    >
      {children}
    </AuthContext.Provider>
  );
};
