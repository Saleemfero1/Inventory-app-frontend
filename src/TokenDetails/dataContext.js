// import React, { createContext, useState } from "react";

// export const DataContext = createContext();

// export const DataProvider = ({ children }) => {
//   const [dashData, setData] = useState([]);

//   const updateData = (newData) => {
//     setData(newData);
//   };

//   return (
//     <DataContext.Provider value={{ dashData, updateData }}>
//       {children}
//     </DataContext.Provider>
//   );
// };
