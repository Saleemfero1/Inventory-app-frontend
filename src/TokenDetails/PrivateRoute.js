import React, { useContext, useState } from "react";
import { Navigate } from "react-router-dom";
import { AuthContext } from "./AuthContext";

const PrivateRoute = ({ children }) => {
  const { token } = useContext(AuthContext);
  return token ? <>{children}</> : <Navigate to="/" />;
};

export default PrivateRoute;
