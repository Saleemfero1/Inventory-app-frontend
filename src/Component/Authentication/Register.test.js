import React from "react";
import { screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { AuthProvider } from "../../TokenDetails/AuthContext";
import { MemoryRouter } from "react-router-dom";
import Register from "./Register";

it("renders correctly", () => {
  const treeOne = renderer
    .create(
      <AuthProvider>
        <MemoryRouter>
          <Register />
        </MemoryRouter>
      </AuthProvider>
    )
    .toJSON();
  expect(screen.getAllByTestId("userName")).toBeInTheDocument();
  expect(treeOne).toMatchSnapshot();
});
