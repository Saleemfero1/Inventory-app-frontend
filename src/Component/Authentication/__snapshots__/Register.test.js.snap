// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="div-body"
>
  <div
    className="div-justify"
  >
    <div
      className="Toastify"
    />
    <main
      className="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
    >
      <div
        className="MuiBox-root css-o4q8eb"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1xvo4iu-MuiAvatar-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LockOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Sign up
        </h1>
        <form
          className="MuiBox-root css-178yklu"
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink={false}
                  htmlFor="username"
                  id="username-label"
                >
                  User Name
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="given-name"
                    autoFocus={true}
                    className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled={false}
                    id="username"
                    name="username"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-yjsfm1"
                    >
                      <span>
                        User Name
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink={false}
                  htmlFor="email"
                  id="email-label"
                >
                  Email Address
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="email"
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled={false}
                    id="email"
                    name="email"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-yjsfm1"
                    >
                      <span>
                        Email Address
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink={false}
                  htmlFor="organizationId"
                  id="organizationId-label"
                >
                  Organization Id
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="Organization"
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled={false}
                    id="organizationId"
                    name="OrganizationId"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-yjsfm1"
                    >
                      <span>
                        Organization Id
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
            >
              <div
                className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink={false}
                  htmlFor="password"
                  id="password-label"
                >
                  Password
                  <span
                    aria-hidden={true}
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="new-password"
                    autoFocus={false}
                    className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled={false}
                    id="password"
                    name="password"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      className="css-yjsfm1"
                    >
                      <span>
                        Password
                         
                        *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-hx3d2o-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            Sign Up
          </button>
          <div
            className="MuiGrid-root MuiGrid-container css-16g5rpk-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              <a
                className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-1cnytyu-MuiTypography-root-MuiLink-root"
                href="/signin"
                onBlur={[Function]}
                onFocus={[Function]}
              >
                Already have an account? Sign in
              </a>
            </div>
          </div>
        </form>
      </div>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-abapg6-MuiTypography-root"
      >
        Copyright © 
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-duj00d-MuiTypography-root-MuiLink-root"
          href="https://mui.com/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          TupleInventory Management
        </a>
         
        2023
        .
      </p>
    </main>
  </div>
</div>
`;
